### PR TITLE
Gracefully handle various errors as NoSuchProcess

### PR DESCRIPTION
**Pull request by mpozniak95**
## Description
This PR converts CalledProcessError during java profiling and FileNotFoundError during ruby profiling to NoSuchProcess exception. 

## Related Issue
https://github.com/Granulate/gprofiler/issues/797

### CODE REVIEW

**Comment by mpozniak95** [comment]
Third point from https://github.com/Granulate/gprofiler/issues/797. 
```
Traceback (most recent call last):
File gprofiler/metadata/application_metadata.py, line 52, in get_metadata
File gprofiler/profilers/java.py, line 316, in make_application_metadata
File granulate_utils/linux/process.py, line 75, in get_mapped_dso_elf_id
File granulate_utils/linux/elf.py, line 42, in inner
File granulate_utils/linux/elf.py, line 28, in inner
File granulate_utils/linux/elf.py, line 98, in get_elf_id
File contextlib.py, line 135, in __enter__
File granulate_utils/linux/elf.py, line 62, in open_elf
FileNotFoundError: [Errno 2] No such file or directory: '/proc/12460/root//usr/lib/jvm/java-1.8.0-amazon-corretto.x86_64/jre/lib/amd64/server/libjvm.so'
```
Will be resolved in separate PR to granulate-utils

File: gprofiler/metadata/versions.py [hunk 1]
```diff
@@ -26,7 +26,14 @@ def get_exe_version(
     def _run_get_version() -> "CompletedProcess[bytes]":
         return run_process([exe_path, version_arg], stop_event=stop_event, timeout=get_version_timeout)
 
-    cp = run_in_ns(["pid", "mnt"], _run_get_version, process.pid)
+    try:
+        cp = run_in_ns(["pid", "mnt"], _run_get_version, process.pid)
+    except FileNotFoundError as e:
+        if not pid_exists(process.pid):
```
Comment (michelhe) [inline gprofiler/metadata/versions.py:32]: Pid recycling can take place before this check.
If you use the is_alive (or whatever its called) method on thr process object you already have, psutil will take care of mitigating pid recycle by also checking if the process start time is the the same

**Comment by mpozniak95** [comment]
Merged master and build-executable passed.

File: gprofiler/profilers/java.py [hunk 2]
```diff
@@ -236,11 +236,16 @@ def __init__(self, stop_event: Event, jattach_timeout: int):
...
-            timeout=self.jattach_timeout,
-        ).stdout.decode()
+        try:
+            return run_process(
+                [asprof_path(), "jcmd", "--jattach-cmd", cmd, str(process.pid)],
+                stop_event=self.stop_event,
+                timeout=self.jattach_timeout,
+            ).stdout.decode()
+        except CalledProcessError as e:
+            if f"Process {process.pid} not found" in str(e):
```
Comment (Jongy) [inline gprofiler/profilers/java.py:246]: I'm not sure this works. The string `Process xx not found` appears in stderr. Is that present in `str(e)`? Perhaps you need `in e.stderr`?

Comment (mpozniak95) [reply gprofiler/profilers/java.py:246]: This should work. I checked it and it is present in str(e):
```
>>> from gprofiler.utils import run_process
>>> from gprofiler.exceptions import CalledProcessError
>>> cmd = ["python", "/home/mpozniak/test.py"]
>>> try:
...   run_process(cmd)
... except CalledProcessError as e:
...   print(str(e))
...
Command ['python', '/home/mpozniak/test.py'] returned non-zero exit status 1.
stdout:
stderr: ERROR!
```

and test.py was just sys.exit("ERROR!")

**Pull request merged**